be1-go/channel/digitalCash/mod_test.go

```diff
@@ -0,0 +1,400 @@
+	require.NoError(t, err)
+
+	require.False(t, channelDC.sockets.Delete("socket"))
+}
+
+// Test that the channel throws an error when it receives an unsubscribe from a
+// non-subscribed source
+func Test_General_Channel_Wrong_Unsubscribe(t *testing.T) {
+	keypair := generateKeyPair(t)
+
+	fakeHub, err := NewfakeHub(keypair.public, nolog, nil)
+	require.NoError(t, err)
+
+	channel := NewChannel("channel0", fakeHub, nolog)
+
+	channelDC, ok := channel.(*Channel)
+	require.True(t, ok)
+
+	file := filepath.Join(protocolRelativePath, "examples", "query", "unsubscribe", "unsubscribe.json")
+	buf, err := os.ReadFile(file)
+	require.NoError(t, err)
+
+	var message method.Unsubscribe
+	err = json.Unmarshal(buf, &message)
+	require.NoError(t, err)
+
+	err = channelDC.Unsubscribe("socket", message)
+	require.Error(t, err, "client is not subscribed to this channel")
+}
+
+// Tests that the channel works correctly when it receives a catchup
+func Test_Consensus_Channel_Catchup(t *testing.T) {
+	// Create the hub
+	keypair := generateKeyPair(t)
+
+	fakeHub, err := NewfakeHub(keypair.public, nolog, nil)
+	require.NoError(t, err)
+
+	// Create the messages
+	numMessages := 5
+	messages := make([]message.Message, numMessages)
+
+	// Create the channel
+	channel := NewChannel("channel0", fakeHub, nolog)
+
+	channelDC, ok := channel.(*Channel)
+	require.True(t, ok)
+
+	for i := 0; i < numMessages; i++ {
+		// Create a new message containing only an id
+		message := message.Message{MessageID: fmt.Sprintf("%d", i)}
+		messages[i] = message
+
+		// Store the message in the inbox
+		channelDC.inbox.StoreMessage(message)
+
+		// Wait before storing a new message to be able to have an unique
+		// timestamp for each message
+		time.Sleep(time.Millisecond)
+	}
+
+	// Compute the catchup method
+	catchupAnswer := channelDC.Catchup(method.Catchup{ID: 0})
+
+	// Check that the order of the messages is the same in `messages` and in
+	// `catchupAnswer`
+	for i := 0; i < numMessages; i++ {
+		require.Equal(t, messages[i].MessageID, catchupAnswer[i].MessageID,
+			catchupAnswer)
+	}
+}
+
+// Tests that the channel throws an error when it receives a publish message
+func Test_General_Channel_Publish(t *testing.T) {
+	// Create the hub
+	keypair := generateKeyPair(t)
+
+	fakeHub, err := NewfakeHub(keypair.public, nolog, nil)
+	require.NoError(t, err)
+
+	// Create the channel
+	channel := NewChannel("channel0", fakeHub, nolog)
+
+	channelDC, ok := channel.(*Channel)
+	require.True(t, ok)
+
+	file := filepath.Join(protocolRelativePath,
+		"examples", "query", "publish", "publish.json")
+	buf, err := os.ReadFile(file)
+	require.NoError(t, err)
+
+	var message method.Publish
+	err = json.Unmarshal(buf, &message)
+	require.NoError(t, err)
+
+	err = channelDC.Publish(message, socket.ClientSocket{})
+	require.Error(t, err, "nothing should be directly published in the general")
+}
+
+// Tests that the channel works correctly when it receives a reaction
```

Comment:
your are only testing the happy path. You should also check what happens with an incorrect message.